source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc

```diff
@@ -30,10 +33,128 @@ namespace Tls {
 
 using SPIFFEConfig = envoy::extensions::transport_sockets::tls::v3::SPIFFECertValidatorConfig;
 
+std::shared_ptr<SpiffeData>
+SPIFFEValidator::parseTrustBundles(const std::string& trust_bundle_mapping_str) {
+  Json::ObjectSharedPtr parsed_json_bundle;
+
+  ENVOY_LOG(info, "Parsing trust_bundles");
+
+  auto json_parse_result = Envoy::Json::Factory::loadFromString(trust_bundle_mapping_str);
+  if (!json_parse_result.ok()) {
+    ENVOY_LOG(error, "Failed to parse SPIFFE bundle map JSON");
+    return nullptr;
+  }
+
+  parsed_json_bundle = json_parse_result.value();
+
+  std::shared_ptr<SpiffeData> spiffe_data = std::make_shared<SpiffeData>();
+
+  const auto trust_domains = parsed_json_bundle->getObject("trust_domains");
+
+  if (!trust_domains.ok() || *trust_domains == nullptr || (*trust_domains)->empty()) {
+    ENVOY_LOG(error, "No trust domains found in SPIFFE bundle map");
+    return nullptr;
+  }
+
+  bool success = true;
+
+  auto status =
+      (*trust_domains)
+          ->iterate([&spiffe_data, &success](const std::string& domain_name,
+                                             const Envoy::Json::Object& domain_object) -> bool {
+            // TODO: Duplicates are currently ignored and only the last value is used.
+            // This is because our json parser auto de-dupes keys in the dict and
+            // only include the last one in this iteration function.
+            spiffe_data->trust_bundle_stores[domain_name] = X509StorePtr(X509_STORE_new());
+
+            ENVOY_LOG(info, "Loading domain '{}' from SPIFFE bundle map", domain_name);
+
+            const auto keys = domain_object.getObjectArray("keys");
+
+            if (!keys.ok() || keys->empty()) {
+              ENVOY_LOG(error, "No keys found in SPIFFE bundle for domain '{}'", domain_name);
+              return (success = false);
```

Comment:
No, please. Let's keep code simple and easy to read, this is key to make the whole repo maintainable.